contribute/testing.md

```diff
@@ -0,0 +1,230 @@
+   </td>
+  </tr>
+  <tr>
+   <td>Triggers
+   </td>
+   <td>Correctness
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/util/TriggerTester.java">TriggerTester</a>
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/windowing/AfterWatermarkTest.java">AfterWatermarkTest</a>
+   </td>
+  </tr>
+  <tr>
+   <td>Coders
+   </td>
+   <td>Encoding/decoding elements
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java">CoderProperties</a>
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/NullableCoderTest.java">NullableCoderTest</a>
+   </td>
+  </tr>
+  <tr>
+   <td>
+   </td>
+   <td>Serialization/deserialization of Coder
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java">CoderProperties</a>
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/NullableCoderTest.java">NullableCoderTest</a>
+   </td>
+  </tr>
+  <tr>
+   <td>
+   </td>
+   <td>Sizing of elements
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java">CoderProperties</a>
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/NullableCoderTest.java">NullableCoderTest</a>
+   </td>
+  </tr>
+  <tr>
+   <td>
+   </td>
+   <td>Deterministic
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java">CoderProperties</a>
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/NullableCoderTest.java">NullableCoderTest</a>
+   </td>
+  </tr>
+  <tr>
+   <td>
+   </td>
+   <td>Structural value equality
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java">CoderProperties</a>
+   </td>
+   <td><a href="https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/test/java/org/apache/beam/sdk/coders/NullableCoderTest.java">NullableCoderTest</a>
+   </td>
+  </tr>
+</table>
+
+
+## Beam Python Testing Matrix
```

Comment:
Let's cut for now. (No documentation for things on feature branches.)

When we do add it, not sure if should be a second table or two new columns in the existing table.